tests/integration/tables/mdfind.cpp

```diff
@@ -29,6 +31,21 @@ TEST_F(Mdfind, test_sanity) {
       "select * from mdfind where query = 'kMDItemFSName = \"*.app\"'"
       " LIMIT 10;");
 
+  // Skip the rest of the assertions if mdfind is disabled. We should still do
+  // the first query though just to be sure osquery doesn't crash in that case.
+  int mdfind_disabled = system("mdutil -s / | grep disabled");
+  if (mdfind_disabled == 0) {
+    LOG(INFO) << "Skipping mdfind test because mdfind is disabled";
+    GTEST_SKIP() << "mdfind is disabled on this system";
+    return;
+  }
+  if (std::string(getenv("GITHUB_JOB")) == "test_older_macos") {
```

Comment:
Careful that `getenv` can return a `nullptr` when the env var doesn't exist, and constructing a `std::string` from a `nullptr` is UB.

Reply:
Good point. Revising.